doc/encoding.rdoc

```diff
@@ -327,24 +304,51 @@ may be specified by @Encoding+Options.
 
 Each of these methods transcodes a string:
 
-String#encode :: Transcodes a string into a new string
-                 according to a given destination encoding,
-                 a given or default source encoding, and encoding options.
+- String#encode: Transcodes +self+ into a new string
+  according to given encodings and options.
+- String#encode!: Like String#encode, but transcodes +self+ in place.
+- String#scrub: Transcodes +self+ into a new string
+  by replacing invalid byte sequences with a given or default replacement string.
+- String#scrub!: Like String#scrub, but transcodes +self in place.
+- String#unicode_normalize: Transcodes +self+ into a new string
+  according to Unicode normalization.
+- String#unicode_normalize!: Like String#unicode_normalize,
+  but transcodes +self+ in place.
+
+=== Transcoding a Stream
+
+Each of these methods may transcode a stream;
+whether it does so depends on the external and internal encodings:
+
+- IO.foreach: Yields each line of given file to the block.
+- IO.new: Creates and returns a new \IO object for the given integer file descriptor.
+- IO.open: Creates a new \IO object.
+- IO.pipe: Creates a connected pair of reader and writer \IO objects.
+- IO.popen: Creates an \IO object to interact with a subprocess.
+- IO.read: Returns a string with all or a subset of bytes from the given file.
+- IO.readlines: Returns an array of strings, which are the lines from the given file.
```

Comment:
I think it's better to use "stream" rather than "file" to be more generic.

```suggestion
- IO.foreach: Yields each line of given stream to the block.
- IO.new: Creates and returns a new \IO object for the given integer file descriptor.
- IO.open: Creates a new \IO object.
- IO.pipe: Creates a connected pair of reader and writer \IO objects.
- IO.popen: Creates an \IO object to interact with a subprocess.
- IO.read: Returns a string with all or a subset of bytes from the given stream.
- IO.readlines: Returns an array of strings, which are the lines from the given stream.
```